src/Common/dataAccess/deleteDatabase.ts

```diff
@@ -33,3 +32,23 @@ export async function deleteDatabase(databaseId: string): Promise<void> {
   clearMessage();
   await refreshCachedResources();
 }
+
+function deleteDatabaseWithARM(databaseId: string): Promise<void> {
+  const subscriptionId: string = userContext.subscriptionId;
```

Comment:
None of these need the type assertion, the compiler can infer it. Same in the other files:

```suggestion
  const subscriptionId = userContext.subscriptionId;
```

Reply:
Done